src/features/imdb.service.js

```diff
@@ -3,19 +3,21 @@ import axios from "axios";
 
 export const fetchAllMovies = createAsyncThunk(

```

Comment:
Ideally the `thunks` should be part of reducer and not the service. The way how we are dividing the reducer layer and service layer is not correct.

Few reasons why:
1. Does it allow you to re-use service in multiple reducers or for multiple use-cases?
2. Does it detaches API calls from redux?

I don't think current implementation answers both of the questions correctly.

This is how it should look like:

1. The reducer file should have thunks
2. The service file should only deal with calling the API, handling API errors and caching data if required.

Eg.

```JSX
// service
const fetchOneMovie = async (id) => {
   const response = await axios.get(....);
   return response.data;
};
export const movieService = { fetchOneMovie };

/// reducer
const fetchOneMovie = createAsyncThunk(
   'movie/fetchOneMovie',
   async (payload) => {
      const movieData = await movieService.fetchOneMovie(payload);
      return movieData;
   }
```